src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,42 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                for stepName in task.listAllStepNames():
+                    sh = task.getStepHelper(stepName)
+                    origCores = max(origCores, sh.getNumberOfCores())
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    origMaxRSS = int(origMaxRSS/1000.) # HTCondor expects MB; we get KB.
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # HTCondor doesn't explicitly scale VSize; it's also not clear what
+                # resources this manages (as we already watch the memory use) or how
+                # it should relate to other resources (such as memory or cores used).
+                # Hence, we simply remove it if we change anything about the memory.
+                if origMaxRSS and (resources['memory'] != origMaxRSS) and ('maxVSize' in args):
+                    del args['maxVSize']
+                # Scale our timeouts according to the scaling of the cores.  *NOTE*
+                # the job ad contains the real scaling here - unfortunately, it uses
+                # ClassAd expressions and we can only parse values.  Something for the
+                # future.
+                # As the scaling is not perfectly linear (efficiency is closer to 95%),
+                # we add a somewhat-arbitrarily-chosen 60 minutes (timeouts are in seconds).
+                coreScaling = resources['cores'] / origCores
```

Comment:
@bbockelm I still would like to see a change where *nothing* gets changed in the watchdog in cases where there is no job resizing. Can you push this `coreScaling = resources['cores'] / origCores` line above and just `continue` the loop in case there was no change?